### ISSUE TITLE
[Teaser] Title editable even though "Get title from linked page" is checked (in certain circumstances)

### ISSUE THREAD

**Issue by PublicisMaikLindner** [adobe/aem-core-wcm-components#694]
## Bug Report

**Current Behavior**
We have an own component, which has the Teaser as super-resource via
```xml
sling:resourceSuperType="core/wcm/components/teaser/v1/teaser"
```

We do not need the cta's, so we hide the properties via
```xml
                   <actions
                        jcr:primaryType="nt:unstructured"
                        sling:resourceType="granite/ui/components/coral/foundation/container">
                        <items jcr:primaryType="nt:unstructured">
                            <columns
                                jcr:primaryType="nt:unstructured"
                                sling:resourceType="granite/ui/components/coral/foundation/fixedcolumns">
                                <items jcr:primaryType="nt:unstructured">
                                    <column
                                        jcr:primaryType="nt:unstructured"
                                        sling:resourceType="granite/ui/components/coral/foundation/container">
                                        <items
                                            jcr:primaryType="nt:unstructured"
                                            sling:hideChildren="[actionsEnabled,actionsDisabled,actions]">
                                        </items>
                                    </column>
                                </items>
                            </columns>
                        </items>
                    </actions>
```

In this case, the checkbox "Get title from linked page" is not evaluated as suggested. When you check this box, click on done and then reedit this teaser-variant, the title is editable:

<img width="502" alt="Bildschirmfoto 2019-08-14 um 13 14 06" src="https://user-images.githubusercontent.com/52775439/63017475-bd313000-be96-11e9-9a47-199ae65ee0af.png">

Any changes are saved, but not clearly visible (because the title is loaded from the linked page).

The interesting thing is, this does not happen, when we don't hide `actions` (via `sling:hideChildren`). So I think there is an unintentional context.

**Expected behavior/code**
The checkbox should work, even if we hide any child-nodes.

**Environment**
- AEM version and patch level: 6.4.2.0
- Core Components version: 2.5.0

**Comment by msagolj** [comment]
@PublicisMaikLindner When you remove the items you list in `sling:hideChildren`, they no longer get rendered and therefore won't get found by the teaser dialog's `dialog-loaded` init function anymore, causing a javascript error:

https://github.com/adobe/aem-core-wcm-components/blob/development/content/src/content/jcr_root/apps/core/wcm/components/teaser/v1/teaser/clientlibs/editor/js/teaser.js#L103

The failing checkbox for `Get title from linked page` is just a side effect oft this javascript error since the code that would update it is not executed anymore.

To proper way to _disable_ the CTA is to go to the content policy and check :

![image](https://user-images.githubusercontent.com/37147400/63093932-3ac47100-bf67-11e9-926e-eb6b48d13357.png)

If you really want to _hide/remove_ the CTA dialog elements you would also have to create your own clientlib and copy/adapt the above code to not fail anymore.

If this answers you question please close the issue.

**Comment by PublicisMaikLindner** [comment]
Thanks @msagolj for the explanation.

The only thing I don't understand is, that I have to adapt the code to not fail anymore. From my perspective there is no reason that the javascript part for the ctas affect that checkbox. So why do you let that code as it is?

**Comment by richardhand** [comment]
@PublicisMaikLindner - I agree, we should make the JS code more robust here to check for the presence of `actions`

**Comment by InsomniacSabbir** [comment]
Hi @richardhand, I would like to work on this issue. I believe nobody is working on this. Can I take this issue?
Thanks.

**Comment by richardhand** [comment]
Fixed in #828

**Comment by richardhand** [comment]
The code for this issue is merged to master and will be available in the upcoming [2.8.0 release](https://github.com/adobe/aem-core-wcm-components/milestone/18).
